rag/llm/key_utils.py

```
@@ -31,4 +34,83 @@ def _normalize_replicate_key(key):
+        return key
+    if isinstance(key, str):
+        try:
+            payload = json.loads(key)
+        except (json.JSONDecodeError, TypeError):
+            logging.warning(
+                'TTS/Seq2txt provider key is not valid JSON; expected a JSON object (e.g. {"fish_audio_ak": "..."} for Fish Audio, or {"spark_app_id": "..."} for XunFei Spark). See conf/llm_factories.json for supported TTS / Seq2txt factories.',
+            )
+            raise ModelException(
+                'TTS/Seq2txt provider key must be a JSON object. Example: {"fish_audio_ak": "..."} or {"spark_app_id": "..."} or {"tencent_cloud_sid": "..."}. See conf/llm_factories.json for supported TTS / Seq2txt factories.',
+                retryable=False,
+            )
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)